app/services/trainees/update.rb

```diff
@@ -4,14 +4,16 @@ module Trainees
   class Update
     include ServicePattern
 
-    def initialize(trainee:, params: {})
+    def initialize(trainee:, params: {}, withdrawal: false)
       @trainee = trainee
       @params = params
+      @withdrawal = withdrawal
     end
 
     def call
       trainee.update!(params)
       Dqt::UpdateTraineeJob.perform_later(trainee)
+      Dqt::WithdrawTraineeJob.perform_later(trainee) if @withdrawal
```

Comment:
Discussed offline. For a withdrawal we don't change any of the params that we include in the update request so we don't need to queue an update job. We're also not sure what happens if these were to be run by sidekiq in the wrong order. For clarity and to avoid that we should have a Trainees::Withdraw service instead of overloading the `Update` one.

Reply:
Thanks @gpeng , I've extracted the withdraw API call into it's own service specifically for the withdrawal feature.